### PR TITLE
add formal page to Other language features

### LINKED ISSUE #22: Formal support is missing in documentation
There is no mentioning in the documentation for the formal support.
It could include things like:
- How to include formal in the design

```
import spinal.core.GenerationFlags._
import spinal.core.Formal._

GenerationFlags.formal{
   assume(... )
   assert(... )
   cover(...)
   ...
}
```

- How to generate a design with formal included
```
val config = SpinalConfig()
        config.includeFormal.generateSystemVerilog(new MyTopLevel())
```
- list the limitation of the current system

### PR DESCRIPTION
**Pull request by piessteady**
closes #22

### CODE REVIEW

**Comment by wifasoi** [comment]
Thank you for your contribution.
I'll review the PR this weekend (sorry but at the moment I'm quite busy).
I gave it a light read and it seamed ok.

**Comment by piessteady** [comment]
You are welcome.
I think it is now ready for some review and feedback.

File: source/SpinalHDL/Other language features/formal.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
+Formal
+==========
+
+General
+-------
+
+There is limited support for SystemVerilog Assertions.
+
+To add Assertions to a Component for formal verification, repeating the practical example from before:
```
Comment (wifasoi) [inline source/SpinalHDL/Other language features/formal.rst]: This sentence doesn't feels right... It feels like there is something missing (`repeating the practical example from before` reference an example that is not there)
maybe:
```suggestion
You can add formal statement (assume, assert ecc..) in the Component definition, like the example below:
```

Comment (piessteady) [reply source/SpinalHDL/Other language features/formal.rst]: I meant the example from the assertion, but your sentence is much clearer.

**Comment by wifasoi** [comment]
LGTM
Thank you for your contribution.

**Pull request merged**